framework/src/org/checkerframework/framework/qual/TargetLocations.java

```diff
@@ -0,0 +1,29 @@
+package org.checkerframework.framework.qual;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * A meta-annotation that indicates locations where a particular qualifier
```

Comment:
There should be a reference to the existing type validation logic and that this is a declarative, coarse-grained approach to achieve the same thing. If more fine-grained control over type.invalid is needed, a TypeValidator should be implemented.